helper/schema/resource_data.go

```diff
@@ -213,9 +212,11 @@ func (d *ResourceData) State() *terraform.InstanceState {
 		}
 
 		raw := d.get([]string{k}, source)
-		rawMap[k] = raw.Value
-		if raw.ValueProcessed != nil {
-			rawMap[k] = raw.ValueProcessed
+		if raw.Exists && !raw.Computed {
```

Comment:
If `raw.Computed` is true it means that the value is actually unknown because it is yet to be computed. It doesn't mean the value itself was computed.